src/Orc.CsvTextEditor/CsvTextEditorTools/Interfaces/ICsvTextEditorTool.cs

```diff
@@ -5,21 +5,22 @@
 // --------------------------------------------------------------------------------------------------------------------
 
 
-namespace Orc.CsvTextEditor.CsvTextEditorToolManagement
+namespace Orc.CsvTextEditor
 {
     using System;
     using ICSharpCode.AvalonEdit;
+    using Services;
 
     public interface ICsvTextEditorTool
     {
         #region Properties
         string Name { get; }
+        bool IsInitialized { get; }
         #endregion
 
-        bool IsInitialized(object scope = null);
         void Open();
         void Close();
-        void Initialize(TextEditor textEditor, object scope = null);
+        void Initialize(TextEditor textEditor, ICsvTextEditorService csvTextEditorService);
```

Comment:
no need to inject service here.
looks like workaround. all services must be injected into ctor or resolved internally